app/models/service/aggregation.rb

```diff
@@ -42,26 +151,36 @@ def aggregate_direct_vm_memory_on_disk
   end
 
   def aggregate_all_vm_cpus
-    all_vms.inject(0) { |aggregate, vm| aggregate += vm.cpu_total_cores.to_i }
+    attributes["aggregate_all_vm_cpus"] || all_vms.inject(0) { |aggregate, vm| aggregate += vm.cpu_total_cores.to_i }
```

Comment:
aaah - got it.

`attributes` creates a new attribute hash ever call. A bit of duplication.
Yes, what you said is true, it will silently fail for an unknown attribute and continue through.

I tend to use `self[attr]` which will not create that hash. And that version will blow up if it doesn't have the attribute.